lib/js/components/DatePickers/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,278 @@
+</style>
+
+<script lang="ts">
+import { toRaw, PropType } from 'vue';
+
+import DsTile from '../../Tile';
+import { ICONS } from '../../Icons/Icon';
+import DatePickerBox from '../DatePickerBox';
+
+import {
+	DATE_PICKER_CALENDAR_POSITIONS,
+	DATE_PICKER_COLORS,
+	DATE_PICKER_STATES,
+	DATE_PICKER_TRIGGER_TYPES,
+	DatePickerCalendarPositions,
+	DatePickerColors,
+	DatePickerStates,
+	DatePickerTriggerTypes,
+} from './DatePicker.consts';
+import { capitalizeFirstLetter } from '../../../../../tools/importers/helpers/modifiers';
+
+let flatpickr: Function | null = null;
+
+const flatpickrPositions = {
+	[DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_LEFT]: 'below left',
+	[DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_RIGHT]: 'below right',
+	[DATE_PICKER_CALENDAR_POSITIONS.BOTTOM]: 'below',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP_LEFT]: 'above left',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP_RIGHT]: 'above right',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP]: 'above',
+};
+export default {
+	name: 'DatePicker',
+	components: {
+		DatePickerBox,
+		DsTile,
+	},
+	props: {
+		triggerType: {
+			type: String as PropType<DatePickerTriggerTypes>,
+			default: DATE_PICKER_TRIGGER_TYPES.TILE,
+		},
+		isInteractive: {
+			type: Boolean,
+			default: true,
+		},
+		placeholder: {
+			type: String,
+			default: 'Wybierz datę',
+		},
+		date: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		label: {
+			type: String,
+			default: '',
+		},
+		isLabelUppercase: {
+			type: Boolean,
+			default: false,
+		},
+		icon: {
+			type: Object,
+			default: null,
+			validate: (icon) => icon === null || Object.values(ICONS).includes(toRaw(icon)),
+		},
+		isIconHiddenOnMobile: {
+			type: Boolean,
+			default: false,
+		},
+		calendarPosition: {
+			type: String as PropType<DatePickerCalendarPositions>,
+			default: DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_LEFT,
+		},
+		errorMessage: {
+			type: String,
+			default: '',
+		},
+		state: {
+			type: (String as PropType<DatePickerStates>) || null,
+			default: DATE_PICKER_STATES.DEFAULT,
+		},
+		color: {
+			type: String as PropType<DatePickerColors>,
+			default: DATE_PICKER_COLORS.NEUTRAL,
+		},
+		defaultDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		disableDates: {
+			type: Array,
+			default: () => [],
+		},
+		minDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		maxDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+	},
+	emits: { 'update:date': () => true },
+	data() {
+		return {
+			datePicker: null,
+			isOpen: false,
+			DATE_PICKER_STATES: Object.freeze(DATE_PICKER_STATES),
+			DATE_PICKER_TRIGGER_TYPES: Object.freeze(DATE_PICKER_TRIGGER_TYPES),
+		};
+	},
+	computed: {
+		isDatePickerEnabled() {
+			return this.isInteractive && this.state === DATE_PICKER_STATES.DEFAULT;
+		},
+		eyebrowText() {
+			if (this.state === DATE_PICKER_STATES.LOADING) {
+				return '';
+			}
+
+			return capitalizeFirstLetter(
+				new Date(this.date).toLocaleDateString(undefined, {
```

Comment:
Haven't you thought about using moment js? we already have it in the platform

Reply:
What it would gave us for the cost of including new, already deprecated, dependency?

Reply:
deprecated? i don't think so, we still use it in platform.
moment is much simpler
```
moment(this.date).format('L');
```
instead of the multi-undefined format

Reply:
This is not a big win for me since we don't use it in DS yet and even its documentation recommends to use other, better libraries
https://momentjs.com/docs/

Reply:
then it's a good time for date-fns :D

Reply:
Maybe. I would discuss pros and cons with the team and write an ADR for that

Reply:
ok, let's leave it as it is - we'll talk about it with the team